v2_compat does not provide shims for ToggleableOutputPin

```rust
error[E0599]: no method named `toggle` found for type `alt_stm32f30x_hal::gpio::PA1<alt_stm32f30x_hal::gpio::PullDown, alt_stm32f30x_hal::gpio::Output<alt_stm32f30x_hal::gpio::PushPull, alt_stm32f30x_hal::gpio::HighSpeed>>` in the current scope
  --> pin_sensors/main.rs:54:21
   |
54 |         let _ = pa1.toggle();
   |                     ^^^^^^
   |
   = note: the method `toggle` exists but the following trait bounds were not satisfied:
           `alt_stm32f30x_hal::gpio::PA1<alt_stm32f30x_hal::gpio::PullDown, alt_stm32f30x_hal::gpio::Output<alt_stm32f30x_hal::gpio::PushPull, alt_stm32f30x_hal::gpio::HighSpeed>> : embedded_hal::digital::v2::ToggleableOutputPin`
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
   |
6  | use embedded_hal::digital::v1::ToggleableOutputPin;

```

Can be solved in hal implementation by providing `v2::toggleable` implementation:

```rust
impl<PT: PullType, OT:OutputType, OS:OutputSpeed> v2::toggleable::Default
    for $PXi<PT, Output<OT, OS>> {}
```

But would be nice to have it provided by this crate.

--- a/src/digital/v2_compat.rs
+++ b/src/digital/v2_compat.rs
@@ -39,6 +39,9 @@ where
      }
 }
 
+#[cfg(feature = "unproven")]
+#[allow(deprecated)]
+impl<T> v2::toggleable::Default for T where T: v1::toggleable::Default {}
 
 /// Implementation of fallible `v2::InputPin` for `v1::InputPin` digital traits
 #[cfg(feature = "unproven")]
@@ -81,6 +84,20 @@ mod tests {
         }
     }
 
+    #[allow(deprecated)]
+    impl v1::StatefulOutputPin for OldOutputPinImpl {
+        fn is_set_low(&self) -> bool {
+            self.state == false
+        }
+
+        fn is_set_high(&self) -> bool {
+            self.state == true
+        }
+    }
+
+    #[allow(deprecated)]
+    impl v1::toggleable::Default for OldOutputPinImpl {}
+
     struct NewOutputPinConsumer<T: v2::OutputPin> {
         _pin: T,
     }
@@ -90,6 +107,25 @@ mod tests {
         pub fn new(pin: T) -> NewOutputPinConsumer<T> {
             NewOutputPinConsumer{ _pin: pin }
         }
+    }
+
+    struct NewToggleablePinConsumer<T: v2::ToggleableOutputPin> {
+        _pin: T,
+    }
+
+    impl<T> NewToggleablePinConsumer<T>
+    where
+        T: v2::ToggleableOutputPin,
+    {
+        pub fn new(pin: T) -> NewToggleablePinConsumer<T> {
+            NewToggleablePinConsumer { _pin: pin }
+        }
+    }
+
+    #[test]
+    fn v2_v1_toggleable_implicit() {
+        let i = OldOutputPinImpl { state: false };
+        let _c = NewToggleablePinConsumer::new(i);
     }
 
     #[test]

